File: api-asic/src/tr/gov/tubitak/uekae/esya/api/asic/core/SignableEntry.java
package tr.gov.tubitak.uekae.esya.api.asic.core;

import tr.gov.tubitak.uekae.esya.api.crypto.alg.DigestAlg;
import tr.gov.tubitak.uekae.esya.api.signature.Signable;
import tr.gov.tubitak.uekae.esya.api.signature.SignatureException;

import java.io.InputStream;

/**
 * @author ayetgin
 */
public class SignableEntry implements Signable
{
    Signable internal;
    String pathInZip;

    public SignableEntry(Signable aInternal, String aPathInZip)
    {
        internal = aInternal;
        pathInZip = aPathInZip;
    }

    public InputStream getContent() throws SignatureException
    {
        return internal.getContent();
    }

    public byte[] getDigest(DigestAlg aDigestAlg) throws SignatureException
    {
        return internal.getDigest(aDigestAlg);
    }

    public String getURI()
    {
        return pathInZip;
    }

    public String getResourceName()
    {
        return internal.getResourceName();
    }

    public String getMimeType()
    {
        return internal.getMimeType();
    }
}

File: api-asic/csharp/src/core/SignableEntry.cs
/**
 * @author yavuz.kahveci
 */

using System.IO;
using tr.gov.tubitak.uekae.esya.api.crypto.alg;
using tr.gov.tubitak.uekae.esya.api.signature;

namespace tr.gov.tubitak.uekae.esya.api.asic.core
{
    public class SignableEntry : Signable
    {
        readonly Signable internalSignable;
        readonly string pathInZip;

        public SignableEntry(Signable aInternalSignable, string aPathInZip)
        {
            internalSignable = aInternalSignable;
            pathInZip = aPathInZip;
        }

        public Stream getContent()
        {
            closeContentStream();
            return internalSignable.getContent();
        }

        public byte[] getDigest(DigestAlg aDigestAlg)
        {
            return internalSignable.getDigest(aDigestAlg);
        }

        public string getURI()
        {
            return pathInZip;
        }

        public string getResourceName()
        {
            return internalSignable.getResourceName();
        }

        public string getMimeType()
        {
            return internalSignable.getMimeType();
        }

        public void closeContentStream()
        {
            internalSignable.closeContentStream();
        }
    }
}
